File: sem1_pmi/programmingbasics/lab5/10.cpp
#include <iostream>

int digit_sum(int);

int main()
{
	for(int i = 105; i < 1000; i += 7)
		if(digit_sum(i) % 7 == 0)
			std::cout << i << ' ';
	std::cout << std::endl;

	return 0;
}

int digit_sum(int n)
{
	if (n < 10) {
		return n;
	}

	return n%10 + digit_sum(n/10);
}

File: sch/exam/10.pas
var i : integer;

function digit_sum(n : integer) : integer;
begin
	if n = 0 then
		digit_sum := 0
	else
		digit_sum := n mod 10 + digit_sum(n div 10);
end;

begin
	for i := 100 to 999 do
		if (i mod 5 = 0) and (digit_sum(i) mod 5 = 0) then
			writeln(i);
end.

